crates/astria-core/src/protocol/fees/v1.rs

```diff
@@ -83,89 +110,79 @@ impl_protobuf_for_fee_components!(
-}
+    #[must_use]
+    pub fn base(&self) -> u128 {
+        self.base
+    }
 
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub struct IbcRelayFeeComponents {
-    pub base: u128,
-    pub multiplier: u128,
+    #[must_use]
+    pub fn multiplier(&self) -> u128 {
+        self.multiplier
+    }
 }
 
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub struct ValidatorUpdateFeeComponents {
-    pub base: u128,
-    pub multiplier: u128,
+impl<T: Protobuf> Debug for FeeComponents<T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.debug_struct(&format!("{}FeeComponents", T::Raw::NAME))
+            .field("base", &self.base)
+            .field("multiplier", &self.multiplier)
+            .finish()
+    }
 }
 
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub struct FeeAssetChangeFeeComponents {
-    pub base: u128,
-    pub multiplier: u128,
+impl Debug for FeeComponents<IbcRelay> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.debug_struct("IbcRelayFeeComponents")
+            .field("base", &self.base)
+            .field("multiplier", &self.multiplier)
+            .finish()
+    }
 }
 
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub struct FeeChangeFeeComponents {
-    pub base: u128,
-    pub multiplier: u128,
+impl<T: ?Sized> Clone for FeeComponents<T> {
+    fn clone(&self) -> Self {
+        *self
+    }
 }
 
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub struct IbcRelayerChangeFeeComponents {
-    pub base: u128,
-    pub multiplier: u128,
-}
+impl<T: ?Sized> Copy for FeeComponents<T> {}
 
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub struct SudoAddressChangeFeeComponents {
-    pub base: u128,
-    pub multiplier: u128,
+impl<T: ?Sized> PartialEq for FeeComponents<T> {
+    fn eq(&self, other: &Self) -> bool {
+        self.base == other.base && self.multiplier == other.multiplier
+    }
 }
 
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub struct IbcSudoChangeFeeComponents {
-    pub base: u128,
-    pub multiplier: u128,
-}
+pub type TransferFeeComponents = FeeComponents<Transfer>;
+pub type RollupDataSubmissionFeeComponents = FeeComponents<RollupDataSubmission>;
+pub type Ics20WithdrawalFeeComponents = FeeComponents<Ics20Withdrawal>;
+pub type InitBridgeAccountFeeComponents = FeeComponents<InitBridgeAccount>;
+pub type BridgeLockFeeComponents = FeeComponents<BridgeLock>;
+pub type BridgeUnlockFeeComponents = FeeComponents<BridgeUnlock>;
+pub type BridgeSudoChangeFeeComponents = FeeComponents<BridgeSudoChange>;
+pub type IbcRelayFeeComponents = FeeComponents<IbcRelay>;
+pub type ValidatorUpdateFeeComponents = FeeComponents<ValidatorUpdate>;
+pub type FeeAssetChangeFeeComponents = FeeComponents<FeeAssetChange>;
+pub type FeeChangeFeeComponents = FeeComponents<FeeChange>;
+pub type IbcRelayerChangeFeeComponents = FeeComponents<IbcRelayerChange>;
+pub type SudoAddressChangeFeeComponents = FeeComponents<SudoAddressChange>;
+pub type IbcSudoChangeFeeComponents = FeeComponents<IbcSudoChange>;
```

Comment:
Yeah - I was mainly trying to reduce churn by providing these aliases, but I was 50-50 myself on adding them :)  Given that you have enough of an opinion on this to have added a comment (even if your opinion is weakly held) I've gone with removing them in [1316e07](https://github.com/astriaorg/astria/pull/1811/commits/1316e076f7b495d2bfa78c8ae9bcea71d6fe97a8) of #1811.  It's a standalone commit, so if others disagree, it can be reverted easily enough.